modules/homa-lite/src/lib.rs

```diff
@@ -283,36 +373,505 @@ pub mod module {
 			Ok(())
 		}
 
-		/// Sets the xcm_dest_weight for XCM transfers.
+		/// Sets the xcm_base_weight for XCM transfers.
 		/// Requires `T::GovernanceOrigin`
 		///
 		/// Parameters:
-		/// - `xcm_dest_weight`: The new weight for XCM transfers.
-		#[pallet::weight(< T as Config >::WeightInfo::set_xcm_dest_weight())]
+		/// - `xcm_base_weight`: The new weight for XCM transfers.
+		#[pallet::weight(< T as Config >::WeightInfo::set_xcm_base_weight())]
 		#[transactional]
-		pub fn set_xcm_dest_weight(origin: OriginFor<T>, xcm_dest_weight: Weight) -> DispatchResult {
+		pub fn set_xcm_base_weight(origin: OriginFor<T>, xcm_base_weight: Weight) -> DispatchResult {
 			T::GovernanceOrigin::ensure_origin(origin)?;
 
-			XcmDestWeight::<T>::put(xcm_dest_weight);
-			Self::deposit_event(Event::<T>::XcmDestWeightSet(xcm_dest_weight));
+			XcmBaseWeight::<T>::put(xcm_base_weight);
+			Self::deposit_event(Event::<T>::XcmBaseWeightSet(xcm_base_weight));
+			Ok(())
+		}
+
+		/// Sets the xcm_unbond_fee for XCM transaction. Fee is paid in Staking Currency.
+		/// Requires `T::GovernanceOrigin`
+		///
+		/// Parameters:
+		/// - `xcm_unbond_fee`: The new fee for XCM transfers.
+		#[pallet::weight(< T as Config >::WeightInfo::set_xcm_unbond_fee())]
+		#[transactional]
+		pub fn set_xcm_unbond_fee(origin: OriginFor<T>, xcm_unbond_fee: Balance) -> DispatchResult {
+			T::GovernanceOrigin::ensure_origin(origin)?;
+
+			XcmUnbondFee::<T>::put(xcm_unbond_fee);
+			Self::deposit_event(Event::<T>::XcmUnbondFeeSet(xcm_unbond_fee));
+			Ok(())
+		}
+
+		/// Put in an request to redeem Staking currencies used to mint Liquid currency.
+		/// The redemption will happen after the currencies are unbonded on the relaychain.
+		///
+		/// Parameters:
+		/// - `liquid_amount`: The amount of liquid currency to be redeemed into Staking currency.
+		/// - `additional_fee`: Percentage of the fee to be awarded to the minter.
+		#[pallet::weight(< T as Config >::WeightInfo::request_redeem())]
+		#[transactional]
+		pub fn request_redeem(origin: OriginFor<T>, liquid_amount: Balance, additional_fee: Permill) -> DispatchResult {
```

Comment:
put this before `mint_for_requests` so later we may want to remove/update set_xxx methods and won't have effect on user facing calls.